### PR TITLE
Organization management

### PR DESCRIPTION
**Pull request by DashaBoichuk**
Added generating password, accounts events.

### CODE REVIEW

File: src/main/java/com/ita/if103java/ims/controller/AccountController.java [hunk 1]
```diff
@@ -1,16 +1,8 @@
...
-import com.ita.if103java.ims.dto.transfer.ExistData;
-import com.ita.if103java.ims.dto.transfer.NewData;
-import com.ita.if103java.ims.entity.User;
 import com.ita.if103java.ims.service.AccountService;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.MediaType;
-import org.springframework.security.core.annotation.AuthenticationPrincipal;
-import org.springframework.validation.annotation.Validated;
 import org.springframework.web.bind.annotation.*;
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/controller/AccountController.java]: Please do not use wildcards in imports
https://stackoverflow.com/questions/3348816/intellij-never-use-wildcard-imports

File: src/main/java/com/ita/if103java/ims/service/impl/AccountServiceImpl.java [hunk 2]
```diff
@@ -40,6 +57,12 @@ public AccountDto view(Long id) {
 
     @Override
     public boolean delete(Long id) {
+        Event event = new Event();
+        event.setMessage("Account was deleted.");
+        event.setAccountId(id);
+        event.setAuthorId(accountDao.findById(id).getAdminId());
+        event.setName(EventName.ORG_DELETED);
+        eventService.create(event);
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/service/impl/AccountServiceImpl.java]: Please create method createEvent in this class and pass there only those arguments which changes.
`event.setAuthorId(accountDao.findById(id).getAdminId());`
Can be select user from the @AuthenticationPrincipal as in methods above

File: src/main/java/com/ita/if103java/ims/service/impl/UpgradeSimpleServiceImpl.java [hunk 3]
```diff
@@ -25,6 +30,13 @@ public void upgradeAccount(Long accountId, Long accountTypeId) {
         Integer newLvl = accountTypeDao.findById(accountTypeId).getLevel();
         if (currentLvl < newLvl) {
             accountDao.upgradeAccount(accountId, accountTypeId);
+
+            Event event = new Event();
+            event.setMessage("Account was upgraded to " + accountTypeDao.findById(accountTypeId).getName() + " level.");
+            event.setAccountId(accountId);
+            event.setAuthorId(accountDao.findById(accountId).getAdminId());
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/service/impl/UpgradeSimpleServiceImpl.java]: Can be select user from the @AuthenticationPrincipal

File: src/main/java/com/ita/if103java/ims/service/impl/UpgradeSimpleServiceImpl.java [hunk 4]
```diff
@@ -25,6 +30,13 @@ public void upgradeAccount(Long accountId, Long accountTypeId) {
         Integer newLvl = accountTypeDao.findById(accountTypeId).getLevel();
         if (currentLvl < newLvl) {
             accountDao.upgradeAccount(accountId, accountTypeId);
+
+            Event event = new Event();
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/service/impl/UpgradeSimpleServiceImpl.java]: @volodymyrvynnyk Could you please create constructor which initializes all below fields, in order to use it different place of the application?

File: src/main/java/com/ita/if103java/ims/controller/UpgradeController.java [hunk 5]
```diff
@@ -21,6 +25,21 @@ public UpgradeController(UpgradeService upgradeService) {
 
     @PutMapping("/{typeId}")
     public void upgrade(@AuthenticationPrincipal User user, @PathVariable("typeId") Long typeId) {
-        upgradeService.upgradeAccount(user.getAccountId(), typeId);
+        upgradeService.upgradeAccount(user, typeId);
+    }
+
+    @GetMapping(value = "/")
+    public AccountTypeDto findCurrentType(@AuthenticationPrincipal User user) {
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/controller/UpgradeController.java]: Just noticed that you use user, but as far as i remember Maria created UserDetailsImpl class which will represent currently logged in user. @MarrMika , correct?
@Mychalevytsh @BohdanAndrusiak @rostIvan @Hvozdianchuck @volodymyrvynnyk @pustoj0 
please pay attention on this, maybe you need to update your classes too.

File: src/main/java/com/ita/if103java/ims/service/impl/InvitationServiceImpl.java [hunk 6]
```diff
@@ -63,6 +61,7 @@ private void sendInvitationMessage(UserDto userDto, Long accountId) {
         mailService.sendMessage(userDto, "Hello, We invite you to join our organization " + account.getName() + " in the Inventory Management System.\n" +
             "Please follow link bellow to proceed with registration:\n" +
             activationURL + userDto.getEmailUUID() + "\n" +
+            "Your password: " + userDto.getPassword() + "\n" +
```
Comment (ikrasWork) [inline src/main/java/com/ita/if103java/ims/service/impl/InvitationServiceImpl.java]: Please add something like this. _For security purpose please change it as soon as possible._

File: src/main/java/com/ita/if103java/ims/controller/UpgradeController.java (hunk 5 above)
Comment (MarrMika) [reply src/main/java/com/ita/if103java/ims/controller/UpgradeController.java]: It is correct. You are right!

**Comment by ikrasWork** [comment]
Please fix conflicts
Conflicting files
pom.xml
src/main/java/com/ita/if103java/ims/mapper/jdbc/AccountRowMapper.java

**Pull request merged**